src/main/scala/com/microsoft/partnercatalyst/fortis/spark/sinks/cassandra/udfs/WeightedAvg.scala

```diff
@@ -44,6 +44,10 @@ abstract class WeightedAvg extends UserDefinedAggregateFunction {
 }
 
 object SentimentWeightedAvg extends WeightedAvg with Serializable{
+  override def evaluate(buffer: Row): Double = {
+    getDoubleValue(Option(buffer.getDouble(0))) / getLongValue(Option(buffer.getLong(1))).toDouble
```

Comment:
I believe this would throw an exception if `getLongValue(Option(buffer.getLong(1))).toDouble` returns 0. I would put a guard against this just to protect against divide by zero.